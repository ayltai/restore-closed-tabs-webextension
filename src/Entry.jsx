import { Avatar, ListItem, ListItemAvatar, ListItemIcon, ListItemText, makeStyles, Tooltip, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const Entry = props => {
    const useStyles = makeStyles(() => ({
        icon    : {
            width  : 16,
            height : 16,
        },
        wrapper : {
            minWidth : 32,
        },
    }));

    const classes = useStyles();

    return (
        <ListItem
            button={props.clickable}
            dense
            disabled={!props.clickable}
            onClick={props.clickable ? props.onClick : undefined}>
            {props.icon && React.isValidElement(props.icon) ? (
                <ListItemIcon className={classes.wrapper}>
                    {props.icon}
                </ListItemIcon>
            ) : (
                <ListItemAvatar className={classes.wrapper}>
                    <Avatar
                        className={classes.icon}
                        variant='square'
                        src={props.icon} />
                </ListItemAvatar>
            )}
            <ListItemText
                disableTypography
                primary={
                    <Tooltip title={props.title}>
                        <Typography noWrap>
                            {props.title}
                        </Typography>
                    </Tooltip>
                } />
        </ListItem>
    );
};

Entry.propTypes = {
    icon      : PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
    ]),
    title     : PropTypes.string.isRequired,
    clickable : PropTypes.bool,
    onClick   : PropTypes.func,
};
