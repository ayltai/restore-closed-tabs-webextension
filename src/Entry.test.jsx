import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Entry } from './Entry';

describe('<Entry />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <Entry
                title='dummy'
                icon={<div />}
                clickable
                onClick={jest.fn()} />
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <Entry
                title='dummy'
                icon={<div />}
                clickable
                onClick={jest.fn()} />
        );
    });

    it('clicks without triggering', () => {
        const onClick = jest.fn();

        shallow(
            <Entry
                title='dummy'
                icon='dummy'
                clickable={false}
                onClick={onClick} />
        ).simulate('click');

        expect(onClick.mock.calls.length).toEqual(0);
    });

    it('triggers onClick when clicked', () => {
        const onClick = jest.fn();

        shallow(
            <Entry
                title='dummy'
                icon='dummy'
                clickable
                onClick={onClick} />
        ).simulate('click');

        expect(onClick.mock.calls.length).toEqual(1);
    });
});
