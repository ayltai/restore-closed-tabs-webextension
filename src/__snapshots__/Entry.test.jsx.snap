// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entry /> renders correctly 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemIcon-root makeStyles-wrapper-2"
  >
    <div />
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-dense"
  >
    <p
      aria-describedby={null}
      className="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="dummy"
    >
      dummy
    </p>
  </div>
  <span
    className="MuiTouchRipple-root"
  />
</div>
`;
